src/client/controller.rs

```diff
@@ -80,12 +80,21 @@ impl ControllerClient {
                 }),
             }
         })
-        .await
+        .await?;
+
+        // Refresh the cache now there is definitely a new topic to observe.
+        let _ = self.brokers.refresh_metadata().await;
```

Comment:
That assumes two things:

- the creator of the topic is actually interested in producing / consuming them, which is NOT true for pure admin clients
- we immediately have usable metadata available. I think for Kafka+Zookeeper the partition leader election will take some time, so it is very likely that the response we cache here doesn't contain any usable information at all.

Overall this seems like a premature optimization. I would rather leave it out and expose some "refresh metadata" call on top-level `Client` that a user CAN call after creating loads of topics and before creating all the clients in hope that this speeds up things (although I'm not really sure that this actually helps).

Reply:
I included it beacuse:

* It is a common pattern to create a topic and immediately initialise clients for it ([we do it!](https://github.com/influxdata/influxdb_iox/blob/b21799acaea38be920b1cc5942a3243abd37dbbf/write_buffer/src/kafka/mod.rs#L64-L78))
* The single extra request is preferable over initialising 100 `PartitionClient` with no cached metadata (if all created quickly they race to populate the cache)
* Producing/consuming clients > admin clients
* I'd hope we can hide the implementation details of caching from the user and not have them know they need to call `refresh()` to avoid a perf footgun

Ultimately I added this because I was creating a topic and initialising 100 `PartitionClient` for it in a local test and saw no performance improvements at all - turns out, it was because the cache wasn't populated with the topic I had just created, so caching was useless! I figured this was relatively common, so stuck this refresh here - I'm not married to it though.